src/RestControllers/FhirPatientRestController.php

```diff
@@ -29,8 +29,29 @@ public function __construct($pid)
         $this->fhirService = new FhirResourcesService();
     }
 
-    // implement put post in future
+	public function post($data){
+        $validationResult = $this->patientService->validate($data);
 
+        $validationHandlerResult = RestControllerHelper::validationHandler($validationResult);
+        if (is_array($validationHandlerResult)) {
+            return $validationHandlerResult; }
+
+        $fhirserviceResult = $this->patientService->insert($data);
```

Comment:
You're treating this as a standard api patient input data structure where it will come to the controller as a FHIR PatientResource json or xml. Otherwise you'd just use the standard api call.
This would then need to be parsed and placed in a data array the patient service will understand. 

Check out https://www.hl7.org/fhir/patient-example.json.html and you can use the R4 classes in src/FHIR/R4 if ya want help with the parse. 

Two ways to go here.
- parse in your REST controller
- add a patient resource parse method to the src/Services/FhirResourcesService.php.

I'd go with latter because it will allow for other resources to build for stack parsing such as with other FHIR resources that are referencing patient resource. Such as say, Observation that may reference patient as part of it's resource.

Reply:
As per my understanding, in case of the 'GET' function in the FHIRPatientRestController, the data which is in json form is converted to FHIR object which is then returned.
Similarly, in case of 'POST' function, is it required to parse the data to FHIR object form. If that's the case then another function for insert will need to be implemented in the PatientService.
Could you please correct me if I am wrong in my understanding here.
Thanks.

Reply:
> in case of the 'GET' function in the FHIRPatientRestController

The patient service returns an array of the patient data where createPatientResource() will convert that data array to a Fhir json object to send back to Fhir aware requesting client.


> in case of 'POST' function, is it required to parse the data to FHIR object form

close. you are now doing reverse of GET. You will need to parse from a Fhir json to an new array for the patient service to handle saves. Patient Service has the insert and update already. You will need to create a method in src/Services/FhirResourcesService.php to parse the Fhir json to the standard patient service data array you will eventually pass to standard PatientService class.

Reply:
I have an example of doing this using api calls in the Publish testing client turned on in globals. Source is at openemr/phpfhir